test/DefaultCluster.Tests/TimerOrleansTest.cs

```diff
@@ -25,89 +25,117 @@ public TimerOrleansTest(ITestOutputHelper output, DefaultClusterFixture fixture)
+            var timeout = period.Multiply(50);
+            while (stopwatch.Elapsed < timeout && last < 10)
+            {
+                await Task.Delay(period.Divide(2));
+                last = await grain.GetCounter();
+            }
 
-            ITimerGrain grain = GrainClient.GrainFactory.GetGrain<ITimerGrain>(GetRandomGrainId());
-            TimeSpan period = grain.GetTimerPeriod().Result;
-            Thread.Sleep(period.Multiply(10));
-            int last = grain.GetCounter().Result;
+            last = await grain.GetCounter();
             output.WriteLine("value = " + last);
-            Assert.True(last >= 10 && last <= 11, "last = " + last.ToString(CultureInfo.InvariantCulture));
+            Assert.True(last >= 10 && last <= 12, $"last >= 10 && last <= 12. Actual: last = {last}");
 
-            var additionalSilo = this.HostedCluster.StartAdditionalSilo();
-            try
-            {
-                //IManagementGrain mgmtGrain = Orleans.Silo.SystemManagement;
-                //mgmtGrain.SuspendHost(Orleans.SiloAddress).Wait();
-
-                Thread.Sleep(period.Multiply(10));
-                grain.StopDefaultTimer().Wait();
-                stopwatch.Stop();
-
-                last = grain.GetCounter().Result;
-                Assert.True(last >= 20);
-                double maximalNumTicks = stopwatch.Elapsed.Divide(grain.GetTimerPeriod().Result);
-                Assert.True(last <= maximalNumTicks);
-                //mgmtGrain.ResumeHost(Orleans.SiloAddress).Wait();
-                output.WriteLine("Total Elaped time = " + (stopwatch.ElapsedMilliseconds / 1000.0) + " sec. Expected Ticks = " + maximalNumTicks + ". Actual ticks = " + last);
-            }
-            finally
+            // Restart the silo.
+            HostedCluster.RestartSilo(HostedCluster.Primary);
+            stopwatch.Restart();
+            period = await grain.GetTimerPeriod();
+
+            // Poke the grain and ensure it still works as it should.
+            last = await grain.GetCounter();
+            while (stopwatch.Elapsed < timeout && last < 10)
             {
-                if (this.HostedCluster.SecondarySilos.Count > 1)
-                {
-                    // do not leave unnecessarily too many silos running
-                    this.HostedCluster.StopSilo(additionalSilo);
-                }
+                await Task.Delay(period.Divide(2));
+                last = await grain.GetCounter();
             }
+
+            last = await grain.GetCounter();
+            Assert.True(last >= 10 && last <= 12, $"last >= 10 && last <= 12. Actual: last = {last}");
+            double maximalNumTicks = stopwatch.Elapsed.Divide(period);
+            Assert.True(
+                last <= maximalNumTicks,
+                $"Assert: last <= maximalNumTicks. Actual: last = {last}, maximalNumTicks = {maximalNumTicks}");
+            //mgmtGrain.ResumeHost(Orleans.SiloAddress).Wait();
```

Comment:
nit: delete this line if you don't need it? 

otherwise LGTM. Thanks for cleaning it. and btw your build failed